src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -187,19 +188,22 @@ class MarathonScheduler @Inject() (
 
   override def error(driver: SchedulerDriver, message: String) {
     log.warn("Error: %s".format(message))
-    suicide()
+    suicide(message)
   }
 
-  private def suicide(): Unit = {
-    log.fatal("Committing suicide")
+  private def suicide(message: String): Unit = {
```

Comment:
Please document this (scaladoc) to make it clear that this method not only shuts down the scheduler but now expunges the framework ID.  In fact, maybe it's better to add a parameter to make that behavior explicit.